dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2159,5 +2159,35 @@ void MergeTreeData::Transaction::replaceParts(MergeTreeData::DataPartState move_
     }
 }
 
+bool MergeTreeData::isIndexColumn(const ASTPtr & node) const
+{
+    String column_name = node->getColumnName();
+
+    for (const auto & column : sort_descr)
+        if (column_name == column.column_name)
```

Comment:
This does not look very efficient, but I am a little unsure if it is worth using a map here. I have found at least 2 places when we construct a new map from the SortDescription columns, which does not look very nice to me as well. I think it might be a good idea to have a pre-constructed column map somewhere to avoid code duplication.

Reply:
That's Ok.
(As this is checked once per query, and primary key is unlikely to be very big).